### PR TITLE
Setting up testing infrastructure

### PR DESCRIPTION
**Pull request by Marketionist**
**This pull request sets up testing infrastructure**

Details:
- adds ```.editorconfig```, ```.jscsrc```, ```.jshintrc```
- updates ```.gitignore```
- updates ```.travis.yml```
- adds ```browserslist```
- adds ```gulpfile.js```
- adds ```package.json```
- updates ```tox.ini```
- adds basic configurations for unit tests and integration tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,16 +18,32 @@ env:
   - TOXENV=py27-dj17-cms31-postgres
   - TOXENV=flake8
 
+ global:
```
Comment (FinalAngel) [inline .travis.yml]: did you use the correct credentials here? you need to create new ones with a new username on saucelabs and code climate

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (vxsx) [inline .jshintrc]: should take jshintrc from here https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/develop/.jshintrc

File: aldryn_newsblog/tests/frontend/unit/test.cl.newsblog.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * @author:    Divio AG
+ * @copyright: http://www.divio.ch
+ */
+
+'use strict';
+/* global Cl, describe, it, expect, beforeEach, afterEach, fixture, spyOn */
+
+// #############################################################################
+// UNIT TEST
+describe('cl.events.js:', function () {
```
Comment (FinalAngel) [inline aldryn_newsblog/tests/frontend/unit/test.cl.newsblog.js]: naming

Comment (Marketionist) [inline aldryn_newsblog/tests/frontend/unit/test.cl.newsblog.js]: Fixed

File: .jshintrc (hunk 2 above)
Comment (Marketionist) [inline .jshintrc]: Updated

File: .travis.yml (hunk 1 above)
Comment (Marketionist) [inline .travis.yml]: Updated

**Comment by FinalAngel** [comment]
codewise LGTM, but donno why travis is not triggered through this PR?

**Comment by Marketionist** [comment]
Maybe it should be merged to get triggered

**Comment by FinalAngel** [comment]
@Marketionist we need to figure out why, @mkoistinen did something disable it?

File: .travis.yml [hunk 4]
```diff
@@ -18,16 +18,32 @@ env:
...
+    - secure: RGXS2qmljsyLVGEifz9yvLwdg33KxPDeuwHx0QFmjSkQuJZP92Xxl36qr6rNel52L+39oya8BGD/ddB6RNI6R5TsoOvQGMlrVZmYy4d8EPA7J8u3+Mpp1zQGK1ACmcnVm+5h8dbHeWtZW/1SovnclUYem16KUmQf35l9G9tKc/I=
+    - secure: Lt1n+6a4hj7BuShOufWfiAF3aTC4B6xpHwwcbklVy+fdy4pkY3okXO9CFtWkOTBRrFrgc8R+PL9dZ6ObK3SE1w7LU3dur89uIvJ4RXQ1uROr40PXAxD8RO+/2wp7nhh6CmG7GEKpuKt5QHbgdFVrJtmyoPp4vco1pU86njtEico=
+    - secure: Jq2JBTBMYLpnrwk6uuAWGOX4y+TEVkBuwSZywSrlKb9sW77DloxRfGXH81Q9juvaAy4KqtcGY6HE5vqJSo77MS/cmbtmZe4ypPKgVFy2YDDUGdLi65vEjkerjAmtS7bYBVur+fWG0mpYAmnfj06d0MgkBgwgCr4yLbJFtW8ElAM=
+
 cache:
   directories:
     - $HOME/.wheelhouse
 
+before_install:
+  - npm install -g npm
```
Comment (mkoistinen) [inline .travis.yml]: I suspect the issue is here, but I'm not confident about this.

Comment (Marketionist) [inline .travis.yml]: It should not be an issue as the same works ok for aldryn-events https://github.com/aldryn/aldryn-events/blob/master/.travis.yml

Comment (Marketionist) [inline .travis.yml]: Looks like the problem was with the ```matrix``` - see https://github.com/aldryn/aldryn-newsblog/pull/264

**Pull request merged**